release.nix

```diff
@@ -1,7 +1,37 @@
 let
-  shared = import ./nix/shared.nix { };
+  hsPkgs = import ./default.nix { };
 in
-  { inherit (shared)
-      hasktorch_cpu
-    ;
-  }
+  {
+  	libtorch = hsPkgs.libtorch-ffi.components.all;
+
+  	hasktorch = hsPkgs.hasktorch.components.all;
+
+  	all-examples = hsPkgs.examples.components.all;
```

Comment:
Unless you have some specific reason, you almost never want to use the `all` component.  Instead, you want to pull out the components that matter to you (usually the library, exes, tests, and benchmarks).  This is because the `all` component is not just a simple `symlinkJoin`, but actually a completely separate derivation that just compiles all of the components in one derivation (similar to how nixpkgs haskell stuff works).  

haskell.nix provides some helper functions to easily pull out components.